focus/focus-centers-element.html

```diff
@@ -0,0 +1,115 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset=utf-8>
+<title>focus() centers element outside displayport</title>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<style>
+
+.container {
+  height: 25vh;
+  width: 100vw;
+  background: yellow;
+  overflow: auto;
+}
+
+.horizontalSpacer {
+  height: 100%;
+  width: 150vw;
+  background: green;
+}
+
+#test0Target {
+  height: 5vh;
+  width: 100%;
+  background: red;
+}
+
+.verticalSpacer {
+  width: 100%;
+  height: 150vh;
+  background: green;
+}
+
+#test1Target {
+  height: 100%;
+  width: 5vw;
+  background: red;
+}
+
+#test1 {
+  white-space: nowrap;
+}
+
+#test1 > div {
+  display: inline-block;
+}
+
+</style>
+</head>
+<body>
+  <div id="test0" class="container">
+    <div class="verticalSpacer"></div>
+    <div tabindex=0 id="test0Target"></div>
+    <div class="verticalSpacer"></div>
+  </div>
+  <br>
+  <div id="test1" class="container">
+    <div class="horizontalSpacer"></div>
+    <div tabindex=1 id="test1Target"></div>
+    <div class="horizontalSpacer"></div>
+  </div>
+</body>
+<script>
+
+function promiseFrame() {
+  return new Promise(resolve => {
+    requestAnimationFrame(() => requestAnimationFrame(() => resolve()));
+  });
+}
+
+promise_test(async (t) => {
+  window.scroll(0, 0);
+  let target = document.getElementById("test0Target");
+
+  // Focus the element and record the scroll position
+  target.focus();
+  await promiseFrame();
+  let focusPos = test0.scrollTop;
+
+  test0.scroll(0, 0);
+
+  // scrollIntoView the element and record the scroll position
+  target.scrollIntoView({block: "center", inline: "center"});
+  await promiseFrame();
+  let scrollPos = test0.scrollTop;
+
+  // Ensure that both scroll positions are within +/- 1
+  assert_approx_equals(focusPos, scrollPos, 1.0,
+                       "focus() is within +/- 1 of a centered scrollIntoView()");
+}, "Element.focus() center in the block direction");
+
+promise_test(async (t) => {
```

Comment:
I'm still confused why there are two tests here. They are identical up until the assert, right? Why not just include both asserts in one test?

Reply:
This was an artifact of test development. Updated to use just the test that asserts both directions are centered.